guide/english/certifications/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/binary-agents/index.md

```diff
@@ -147,6 +147,24 @@ Make sure that each time you transcode a character from binary to decimal, you r
 
 *   <a href='http://forum.freecodecamp.com/t/javascript-array-prototype-map/14294' target='_blank' rel='nofollow'>Array.prototype.map</a>
 
+## ![:rotating_light:](https://forum.freecodecamp.com/images/emoji/emoji_one/rotating_light.png?v=3 ":rotating_light:") Alternative Advanced Code Solution:
+```js
+  const binaryAgent = str => str.replace(/\d+./g, char => String.fromCharCode(`0b${char}`));
+       
+  binaryAgent("01000001 01110010 01100101 01101110 00100111 01110100 00100000 01100010 01101111 01101110 01100110 01101001 01110010 01100101 01110011 00100000 01100110 01110101 01101110 00100001 00111111");
+```
+
+## Code Explanation
+
+*   Find all groups of one digit followed by one other character
+*   Replace with a string created from the specified sequence of UTF-16 code units
```

Comment:
I think you need to give a bit more information on what the significance of the `0b` in the template literal `0b${char}` is.

Reply:
Thanks for the feedback. Commit 059be86 explains the significance and links to MDN's article on numerical literals.

Reply:
True, but it would be great if you could expand your explanation to how it works. Otherwise, it not obvious why you have to add the `b0` to the front.